oss-internship-2020/libarchive/ld_preload_example/sapi_minitar.cc

```diff
@@ -0,0 +1,207 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "sapi_minitar.h"  // NOLINT(build/include)
+
+#include <dlfcn.h>
+
+#include <iostream>
+#include <memory>
+
+#include "libarchive_sapi.sapi.h"  // NOLINT(build/include)
+#include "sandbox.h"               // NOLINT(build/include)
+#include "sandboxed_api/var_array.h"
+#include "sandboxed_api/var_ptr.h"
+
+std::unique_ptr<SapiLibarchiveSandboxExtract> sandbox_extract;
+std::unique_ptr<LibarchiveApi> api;
+
+struct ExtractTempDirectoryCleanup {
+  ExtractTempDirectoryCleanup(const std::string& dir) : dir_(dir) {}
+  ~ExtractTempDirectoryCleanup() {
+    sandbox2::file_util::fileops::DeleteRecursively(dir_);
+  }
+
+ private:
+  std::string dir_;
+};
+
+std::unique_ptr<ExtractTempDirectoryCleanup> cleanup_ptr;
```

Comment:
No need for a `unique_ptr` here. And also not allowed by the style guide. to reliably clean up after the sandbox exists, you'll need to remove the temp directories from the host code.